src/liblzma/meson.build

```diff
@@ -155,26 +155,18 @@ if ('sha256' in get_option('checks')) and (not get_option('external-sha256'))
     ]
 endif
 
-if ('lzma1' in get_option('encoders')) or
-   ('lzma1' in get_option('decoders')) or
-   ('lzma2' in get_option('encoders')) or
-   ('lzma2' in get_option('decoders'))
-
-    # lz/Makefile.inc
-    if ('lzma1' in get_option('encoders')) or
-       ('lzma2' in get_option('encoders'))
-        lzma_src += [
-            'lz/lz_encoder.c',
-            'lz/lz_encoder_mf.c'
-        ]
-    endif
+# lz/Makefile.inc
+if ('lzma1' in get_option('encoders')) or ('lzma2' in get_option('encoders'))
+    lzma_src += [
+        'lz/lz_encoder.c',
+        'lz/lz_encoder_mf.c'
+    ]
+ endif
 
-    if ('lzma1' in get_option('encoders')) or
-       ('lzma2' in get_option('encoders'))
-        lzma_src += [
-            'lz/lz_decoder.c'
-        ]
-    endif
+if ('lzma1' in get_option('decoders')) or ('lzma2' in get_option('decoders'))
+    lzma_src += [
+        'lz/lz_decoder.c'
+    ]
 endif
 
 if ('lzma1' in get_option('encoders')) or ('lzma1' in get_option('decoders'))
```

Comment:
I wonder if things like this should be factored out into their own variable, like:
```meson
have_lzma1_encoder = ...
```

Reply:
i wonder if lzma1 and lzma2 should be not an option, but just required, as it's the main purpose of xz. That would simplify the build a LOT...

Reply:
As discussed on IRC, it's fine to make LZMA1 and LZMA2 mandatory. Having a way to disable encoder support entirely is useful though.